src/lib/keyed_container.js

```diff
@@ -61,7 +64,11 @@ module.exports = function keyedContainer(baseObj, path, keyName, valueName) {
                 nestedProperty(newValue, valueName).set(value);
             }
 
-            arr[idx] = newValue;
+            // If it's not an unset, force that bit to be unset. This is all related to the fact
+            // that undefined and null are a bit specially implemented in nestedProperties.
+            if(value !== null) {
+                changeType = changeType & ~UNSET;
```

Comment:
I'm tracking what needs to be updated with a bit mask (which, before adding a bit to track whether it's unset in a way that `nestedProperty` needs to handle specially, originally made 10% more sense since everything else was strictly an OR operation). & ~4 unsets precisely the 4 bit.